streaming/index.js

```diff
@@ -622,40 +622,61 @@ const startServer = async () => {
 
     log.verbose(req.requestId, `Starting stream from ${ids.join(', ')} for ${accountId}`);
 
-    // Currently message is of type string, soon it'll be Record<string, any>
-    const listener = message => {
-      const { event, payload, queued_at } = message;
+    const transmit = (event, raw_payload, filter_results) => {
+      let payload = raw_payload;
+      if ((event === 'update' || event === 'status.update') && filter_results && filter_results.length > 0) {
+        // copy the payload and add the `filtered` attribute, an Array of FilterResult
+        // https://docs.joinmastodon.org/entities/Status/#filtered
+        payload = {
+          ...payload,
+          filtered: filter_results
+        };
+      }
 
-      const transmit = (payload) => {
-        const now = new Date().getTime();
-        const delta = now - queued_at;
-        const encodedPayload = typeof payload === 'object' ? JSON.stringify(payload) : payload;
+      // TODO: Replace "string"-based delete payloads with object payloads:
+      const encodedPayload = typeof payload === 'object' ? JSON.stringify(payload) : payload;
 
-        log.silly(req.requestId, `Transmitting for ${accountId}: ${event} ${encodedPayload} Delay: ${delta}ms`);
-        output(event, encodedPayload);
-      };
+      log.silly(req.requestId, `Transmitting for ${accountId}: ${event} ${encodedPayload}`);
+      output(event, encodedPayload);
+    };
 
-      // Only messages that may require filtering are statuses, since notifications
-      // are already personalized and deletes do not matter
-      if (!needsFiltering || event !== 'update') {
-        transmit(payload);
+    // Currently message is of type string, soon it'll be Record<string, any>
+    const listener = message => {
+      const { event, payload } = message;
+
+      // Streaming only needs to apply filtering to some channels and only some
+      // of the events on those channels require filtering. This is because
+      // majority of the filtering happens on the Ruby on Rails (mastodon-web)
+      // side when producing the event for streaming.
+      //
+      // Specifically, the only events that require filtering are `update` and
+      // `status.update`, all other events are transmitted to the client as soon
+      // as they're received (pass-through).
+      if (!needsFiltering || (event !== 'update' && event !== 'status.update')) {
```

Comment:
@ClearlyClaire can you double check my boolean logic here? I think I got it right for what we want:

> If the stream doesn't need filtering, or the event isn't a `update` or `status.update` event, then transmit it directly.

Reply:
That looks fine.